torch/csrc/jit/codegen/cuda/transform_replay.cpp

```diff
@@ -286,7 +286,8 @@ std::pair<TensorDomain*, unsigned int> TransformReplay::replayPasC(
     auto it = replay_PasC.getReplay().find(c_id);
     if (it == replay_PasC.getReplay().end()) {
       TORCH_INTERNAL_ASSERT(
-          c_id->isBroadcast() || c_id->isGather() || c_id->isVectorComponent(),
+          c_id->isBroadcast() || c_id->isGather() ||
+              c_id->isVectorComponent() || is_producer_lookup_tv,
```

Comment:
Allowing any `c_id` when the producer is an input to `index_select` doesn't seem right. We should check if `c_id` depends on the selected dimension. 

You could do something like:

```
IterDomain* consumer_root_selected_domain = ...;
bool c_id_depends_on_selectd_domain = DependencyCheck::isDependencyOf(consumer_root_selected_domain, c_id);
...
TORCH_INTERNAL_ASSERT(
          c_id->isBroadcast() || c_id->isGather() ||
              c_id->isVectorComponent() || c_id_depends_on_selectd_domain,
```

Reply:
you're right. Fixed, see https://github.com/csarofeen/pytorch/pull/2183/files#diff-3051147c9c0739bc571fb8109b8333e682245de4b0aaf8e5e18d6a3419c43c51R287-R294